metadata-ingestion/setup.py

```diff
@@ -23,7 +23,7 @@ def get_long_description():
 base_requirements = {
     # Compatability.
     "dataclasses>=0.6; python_version < '3.7'",
-    "typing_extensions>=3.10.0.2,<4",
+    "typing_extensions>=3.10.0.2",
```

Comment:
You may want to cross-check this. I think this restriction was added for google cloud composer compatibility when one of the customers said our airflow package was not installing

Reply:
I checked before and it seems like the latest composer uses the same typing-> https://cloud.google.com/composer/docs/concepts/versioning/composer-versions

Reply:
We really need a way to have different releases for different airflow versions. Someone might still be using an older airflow version and we would be back to compatibility issues. 
But yeah that can be done separately.